### PR TITLE
feat: authentication

### PR DESCRIPTION
**Pull request by shraddha15-creator**
Added Authentication

### CODE REVIEW

**Comment by veereshbv04** [comment]
Hi, I guess there is error while logging and sign up. Check whether you have pushed the latest commits
image

**Review by MonikaJhingan — APPROVED** [review]
The ckde looks good to be merged.

File: src/components/Card/ProductCard.js [hunk 1]
```diff
@@ -37,15 +40,21 @@ const ProductCard = ({
...
 						<button className="btn btn-dark">Go to Cart</button>
 					</Link>
 				) : (
-					<button className="btn btn-dark" onClick={addToCart}>
+					<button
+						className="btn btn-dark"
+						// onClick={addToCart}
+						onClick={() => {
+							user.token ? addToCart() : navigate("/login");
+						}}
```
Comment (amartya-rs) [inline src/components/Card/ProductCard.js:56]: suggestion of  checking for `user.token` you can have a state(boolean) `isLoggedIn`

File: src/App.js [hunk 2]
```diff
@@ -20,8 +21,23 @@ function App() {
 					<Route path="/products" element={<Products />} />
 					<Route path="/login" element={<Login />} />
 					<Route path="/signup" element={<Signup />} />
-					<Route path="/wishlist" element={<Wishlist />} />
-					<Route path="/cart" element={<Cart />} />
+					{/* private route */}
+					<Route
+						path="/wishlist"
```
Comment (amartya-rs) [inline src/App.js:26]: For a cleaner implementation of private route you can check out `outlets` feature of react router.
https://reactrouter.com/docs/en/v6/getting-started/concepts#outlets

Comment (shraddha15-creator) [reply src/App.js:26]: sure thank you for suggestions

File: src/components/Card/ProductCard.js (hunk 1 above)
Comment (shraddha15-creator) [reply src/components/Card/ProductCard.js:56]: thanks for pointing out

**Comment by shraddha15-creator** [comment]
> Hi, I guess there is error while logging and sign up. Check whether you have pushed the latest commits image

yeah i forgot to add env variables to netlify, thanks for pointing it out

**Pull request merged**